egs/swbd/s5c/local/reverberate/reverberate_data_dir.py

```diff
@@ -0,0 +1,165 @@
+    parser.add_argument("--noise-list", type=str, default = None,
+                        help="Noise information file")
+    parser.add_argument("--num-replications", type=int, dest = "num_replica", default = 1,
+                        help="Number of replicate to generated for the data")
+    parser.add_argument("input_dir",
+                        help="Input data directory")
+    parser.add_argument("output_dir",
+                        help="Output data directory")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.output_dir):
+        os.makedirs(args.output_dir)
+
+    ## Check arguments.
+    if args.rir_list is None:
+        raise Exception("Rir information file must be provided")
+    
+    if not os.path.isfile(args.rir_list):
+        raise Exception(args.rir_list + "not found")
+    
+    if args.noise_list is not None:
+        if not os.path.isfile(args.noise_list):
+            raise Exception(args.noise_list + "not found")
+
+    return args
+
+
+def Corrupt_wav(input_dir, output_dir, rir_list, num_replica, prefix):
+    rirs = list_cyclic_iterator(return_nonempty_lines(open(rir_list).readlines()), random_seed = 1)
+    wav_files = open(input_dir + "/wav.scp", 'r').readlines()
+    command_list = []
+    for i in range(num_replica):
+        for j in range(len(wav_files)):
+            split1 = wav_files[j].split()
+            wav_pipe = " ".join(split1[1:])
+            output_wav_id = prefix + str(i) + "_" + split1[0]
+            rir = rirs.next().split()
+            rir_file_location = rir[-1]
+            command_list.append("{2} {0} wav-reverberate - {1} - |\n".format(wav_pipe, rir_file_location, output_wav_id))
+
+    file_handle = open(output_dir + "/wav.scp", 'w')
+    file_handle.write("".join(command_list))
+    file_handle.close()
+
+
+def Corrupt_wav_noise(input_dir, output_dir, rir_list, noise_list, snr_array, num_replica, prefix):
+    rirs = list_cyclic_iterator(return_nonempty_lines(open(rir_list).readlines()), random_seed = 1)
+    noises = list_cyclic_iterator(return_nonempty_lines(open(noise_list).readlines()), random_seed = 1)
+    snrs = list_cyclic_iterator(snr_array)
+    wav_files = open(input_dir + "/wav.scp", 'r').readlines()
+    command_list = []
+    for i in range(num_replica):
+        for j in range(len(wav_files)):
+            split1 = wav_files[j].split()
+            wav_pipe = " ".join(split1[1:])
+            output_wav_id = prefix + str(i) + "_" + split1[0]
+            rir = rirs.next().split()
+            rir_file_location = rir[-1]
+            noise = noises.next().split()
+            noise_file_location = noise[-1]
+            snr = snrs.next()
+            command_list.append("{4} {0} wav-reverberate --noise-file={2} --snr-db={3} - {1} - |\n".format(wav_pipe, rir_file_location, noise_file_location, snr, output_wav_id))
```

Comment:
I think we would have to decide how these noises and rirs are chosen. e.g. Is a isotropic noise always chosen if it exists? How many point source noises or rirs are chosen for each file?
We should be choosing a foreground noise and background noise separately. Foreground noise is usually short non-stationary noise like door knocks, taps, siren etc. It would not make sense to add these in the same way as the stationary background noise, which I believe is just added over the entire duration of the file. The way I was adding them is that wav-reverberate takes a bunch of several different foreground noises and randomly adds them to the signal at different times.